src/main/java/io/debezium/connector/jdbc/type/debezium/VariableScaleDecimalType.java

```diff
@@ -36,8 +37,11 @@ public String[] getRegistrationKeys() {
     public String getTypeName(DatabaseDialect dialect, Schema schema, boolean key) {
         // The data passed by VariableScaleDecimal data types does not provide adequate information to
         // resolve the precision and scale for the data type, so instead we're going to default to the
-        // maximum double-based data types for the dialect, using DOUBLE.
-        return dialect.getTypeName(Types.DOUBLE);
+        // maximum for the dialect.
+        // TODO change this after https://issues.redhat.com/browse/DBZ-6836 is implemented
+        int precision = Math.min(dialect.getDoublePrecision(), dialect.getDefaultDecimalPrecision());
+        int scale = Math.min(dialect.getMaxScale(), precision);
+        return dialect.getTypeName(Types.NUMERIC, Size.precision(precision, scale));
```

Comment:
After some tests I found that this is not correct. Setting scale equals to precision means that no space will be left for number before decimal dot. 

So in this case this [DBZ-6836](https://issues.redhat.com/browse/DBZ-6836) is no more optional but required for this fix.